backend/routes/ridesharingRoutes.js

```diff
@@ -583,4 +443,201 @@ router.patch('/modify/coordinates/datetime', async(req, res) => {
   res.json(ridesharing)
 })
 
+// Get recommendations
+router.get('/recommendations/:eventId/:userId', async (req, res) => {
+  const { eventId, userId } = req.params
+  let rideshares = []
+  let attendance = []
+  let atendanceId = []
+
+  let event = []
+
+  let distanceRideshareToUser = []
+  let differenceBetweenDates = []
+
+  let userToRecommend = []
+  let userToRecommendAttendance = []
+  let userToRecommendPreferences = []
+  // Part 1 ***********************************************************
+  // Find our user data
+  try {
+    userToRecommend = await prisma.user.findUnique({
+      where: { 
+        id: parseInt(userId)
+      }
+    })
+  } catch (error) {
+    console.log(error);
+
+    return res.status(500).json({
+      success: false,
+      message: error.message,
+    });
+  }
+
+  userToRecommendAttendance = await prisma.attendance.findFirst({
+    where: { 
+      eventId: parseInt(eventId),
+      userAttending: userToRecommend.username,
+    }
+  })
+
+  userToRecommendPreferences = await prisma.ridesharingUserPreferencesForEvent.findUnique({
+    where: { 
+      attendanceId: userToRecommendAttendance.id
+    }
+  })
+  // -----------------------------------------------
+  // Find the event data
+  event = await prisma.event.findMany({
+    where: { 
+      id: parseInt(eventId)
+    }
+  })
+  // ------------------------------------------------
+  // Find the "attendance" of the users that are driving to the event
+  try {
+      attendance = await prisma.attendance.findMany({
+        where: { 
+          eventId: parseInt(eventId),
+          ridesharingDriver: {isNot: null}
+        }
+      })
+    } catch (error) {
+      console.log(error);
+  
+      return res.status(500).json({
+        success: false,
+        message: error.message,
+      });
+  }
+  // ------------------------------------------------
+  // Store the attendance ids in an array
+  attendance.map((a) => {
+    atendanceId.push(a.id)
+  })
+  // ------------------------------------------------
+  // Search the rideshares
+  try {
+    rideshares = await prisma.ridesharing.findMany({
+        where: {
+            attendanceId: { in: atendanceId }
+        }
+    })
+  } catch (error) {
+    console.log(error);
+
+    return res.status(500).json({
+      success: false,
+      message: error.message,
+    });
+  }
+  // -----------------------------------------------
+  // Part 1 ***********************************************************
+
+  // Part 2 ***********************************************************
+  let rideshareScore = []
+
+  // We store the distance between rideshare and user
+  let distance = 0
+  let distanceEventUser = ((event.lat - userToRecommend.lat)**2 + (event.long - userToRecommend.long)**2)**.5
+  rideshares.map((rideshare) => {
```

Comment:
This is Euclidean distance. For route planning, can we be more accurate?

Reply:
Solved in PR #56 by calculating the distance using the Distance Matrix API